src/viam/ur/module/ur_arm.cpp

```diff
@@ -508,6 +522,18 @@ void URArm::keep_alive() {
     VIAM_SDK_LOG(info) << "keep_alive thread terminating";
 }
 
+// https://github.com/UniversalRobots/Universal_Robots_Client_Library/blob/bff7bf2e2a85c17fa3f88adda241763040596ff1/include/ur_client_library/ur/datatypes.h#L204
+std::string URArm::model_type() {
+    std::string model = model_.to_string();
+    if (model == "viam:universal-robots:ur5e") {
+        return "UR5";
+    }
+    if (model == "viam:universal-robots:ur20e") {
+        return "UR20";
+    }
+    return "UNRECOGNIZED_MODEL_TYPE:" + model;
```

Comment:
- Add a check to the constructor that the `Model` argument is a supported one (I should have done this), and throw std::invalid_argument if it is not.
- Throw std::logic_error from this function if somehow we managed to get into this without a valid model, rather than making a non-meaningful return value.

Reply:
moved the logic for getting the model type to the top of the constructor to throw the invalid argument error